Purchases/Logging/Logger.swift

```diff
@@ -71,24 +77,61 @@ extension Logger {
         log(level: .error, intent: .appleError, message: message)
     }
 
+    static func appleError(_ message: CustomStringConvertible) {
+        log(level: .error, intent: .appleError, message: message.description)
+    }
+
     static func appleWarning(_ message: String) {
         log(level: .warn, intent: .appleError, message: message)
     }
 
+    static func appleWarning(_ message: CustomStringConvertible) {
+        log(level: .warn, intent: .appleError, message: message.description)
+    }
+
     static func purchase(_ message: String) {
         log(level: .debug, intent: .purchase, message: message)
     }
 
+    static func purchase(_ message: CustomStringConvertible) {
+        log(level: .debug, intent: .purchase, message: message.description)
+    }
+
     static func rcPurchaseSuccess(_ message: String) {
         log(level: .info, intent: .rcPurchaseSuccess, message: message)
     }
 
+    static func rcPurchaseSuccess(_ message: CustomStringConvertible) {
+        log(level: .info, intent: .rcPurchaseSuccess, message: message.description)
+    }
+
     static func rcSuccess(_ message: String) {
         log(level: .debug, intent: .rcSuccess, message: message)
     }
 
+    static func rcSuccess(_ message: CustomStringConvertible) {
+        log(level: .debug, intent: .rcSuccess, message: message.description)
+    }
+
     static func user(_ message: String) {
         log(level: .debug, intent: .user, message: message)
     }
 
+    static func user(_ message: CustomStringConvertible) {
+        log(level: .debug, intent: .user, message: message.description)
+    }
+
+}
+
+private extension Logger {
+
+    static func log(level: LogLevel, message: String) {
+        guard self.logLevel.rawValue <= level.rawValue else { return }
+        logHandler(level, message)
+    }
+
+    static func log(level: LogLevel, intent: LogIntent, message: String) {
+        let messageWithPrefix = "\(intent.suffix) \(message)"
```

Comment:
`intent.suffix` being used as a prefix. There's something here we should refactor 😄

Reply:
yeah, it was never really a suffix, I'll update.